src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK001DeriveFromAsyncPackageCodeFix.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+
+namespace Microsoft.VisualStudio.SDK.Analyzers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Simplification;
+
+    /// <summary>
+    /// Offers code fixes for diagnostics produced by the <see cref="VSSDK001DeriveFromAsyncPackageAnalyzer"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class VSSDK001DeriveFromAsyncPackageCodeFix : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> ReusableFixableDiagnosticIds = ImmutableArray.Create(
+          VSSDK001DeriveFromAsyncPackageAnalyzer.Descriptor.Id);
+
+        /// <inheritdoc />
+        public override ImmutableArray<string> FixableDiagnosticIds => ReusableFixableDiagnosticIds;
+
+        /// <inheritdoc />
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var baseTypeSyntax = root.FindNode(diagnostic.Location.SourceSpan).FirstAncestorOrSelf<BaseTypeSyntax>();
+            var classDeclarationSyntax = baseTypeSyntax.FirstAncestorOrSelf<ClassDeclarationSyntax>();
+
+            context.RegisterCodeFix(
+                CodeAction.Create("Change base type to AsyncPackage", ct => this.ChangeBaseTypeAsync(context.Document, diagnostic, ct), classDeclarationSyntax.Identifier.ToString()),
+                diagnostic);
+        }
+
+        /// <inheritdoc />
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        private async Task<Document> ChangeBaseTypeAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Can extension developers enable these analyzers on VS 2013 projects without referecing Shell.14.0 or Shell.15.0 in which case this fix or the original warning wouldn't apply to them?

Reply:
Yes they can, and I haven't accommodated for that yet.

Reply:
Yeah, this one is interesting to me too. If a vsix project targets VS 2013 or below, this diagnostic shouldn't apply. They can choose to simply ignore the diagnostic but I wonder if it's possible to be smarter. Maybe looking at the references of the current project and determining if the version of Shell they're referencing is Shell.14+.

Reply:
There is a solution, yes. We could look at references but that would get tedious to maintain long-term and isn't a great pattern IMO. Instead, I plan to support this by testing for the existence of the type within that compilation unit. If it exists, the analyzer will run -- otherwise it will turn itself off. This should scale and be a pattern that other analyzers can use too.